### PR TITLE
Simplify blockchain configuration [ECR-3357]

### PR DESCRIPTION
**Pull request by alekseysidorov**
This PR removes inconvenient `StoredConfiguration` and `GenesisConfig` structures and replaces them by the single `ConsensusConfig` with the protobuf specification.
Also PR removes methods like `following_configuration` and simplifies blockchain::Schema in general.

### CODE REVIEW

File: exonum/src/sandbox/consensus/config.rs [hunk 1]
```diff
@@ -34,90 +33,20 @@ fn test_exclude_validator_from_consensus() {
...
-        prev_cfg
-    );
-    assert_eq!(
-        Schema::new(&sandbox.blockchain().snapshot())
-            .configuration_by_height(sandbox.current_height()),
-        following_cfg
-    );
-}
+// - include validator to consensus
+// TODO
```
Comment (popzxc) [inline exonum/src/sandbox/consensus/config.rs]: ECR for todo?

File: exonum/src/sandbox/mod.rs [hunk 2]
```diff
@@ -436,11 +435,7 @@ impl Sandbox {
     }
 
     pub fn validators(&self) -> Vec<PublicKey> {
-        self.cfg()
-            .validator_keys
-            .iter()
-            .map(|x| x.consensus_key)
-            .collect()
+        self.cfg().validators.iter().map(|x| x.consensus).collect()
```
Comment (popzxc) [inline exonum/src/sandbox/mod.rs]: On one hand `ValidatorKeys` structure obviously contains keys and names of inner fields like `consensus_key` seemed redundant, but however `x.consensus` looks strange.

File: exonum/src/blockchain/config.rs [hunk 3]
```diff
@@ -80,8 +67,11 @@ pub struct StoredConfiguration {
 ///
 /// For additional information on the Exonum consensus algorithm, refer to
 /// [Consensus in Exonum](https://exonum.com/doc/version/latest/architecture/consensus/).
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, ProtobufConvert)]
+#[exonum(crate = "crate", pb = "blockchain::Config")]
```
[the remark this answers is not in the document]
Comment (alekseysidorov) [reply exonum/src/blockchain/config.rs]: `exonum` means only that attribute belongs to Exonum derives.

[the remark this answers is not in the document]
Comment (popzxc) [reply exonum/src/blockchain/config.rs]: We'll this fact is an implementation detail, am I wrong?

File: exonum/src/blockchain/schema.rs [hunk 4]
```diff
@@ -305,75 +296,21 @@ where
...
-            .unwrap_or_else(|| panic!("Configuration at index {} not found", idx));
-        let cfg_hash = cfg_ref.cfg_hash();
-        self.configuration_by_hash(cfg_hash)
-            .unwrap_or_else(|| panic!("Config with hash {:?} is absent in configs table", cfg_hash))
-    }
-
-    /// Returns the configuration for the given configuration hash.
-    pub fn configuration_by_hash(&self, hash: &Hash) -> Option<StoredConfiguration> {
-        self.configs().get(hash)
+    pub fn consensus_config(&self) -> ConsensusConfig {
```
[the remark this answers is not in the document]
Comment (alekseysidorov) [reply exonum/src/blockchain/schema.rs]: Yes, I do not know even a single example where it was needed. Config history in old implementation was written before that we implemented `before_commit` feature. And history was use to determine an actual configuration for the current blockchain height.

File: exonum/src/helpers/fabric/details.rs [hunk 5]
```diff
@@ -0,0 +1,960 @@
...
+                false,
+                "Listen address for private api.",
+                None,
+                "private-api-address",
+                false,
+            ),
+            Argument::new_named(
+                CONSENSUS_KEY_PASS_METHOD,
+                false,
+                "Passphrase entry method for consensus key.\n\
```
[the remark this answers is not in the document]
Comment (alekseysidorov) [reply exonum/src/helpers/fabric/details.rs]: Heh, fabric killed by @vitvakatu

File: exonum/src/sandbox/mod.rs (hunk 2 above)
Comment (alekseysidorov) [reply exonum/src/sandbox/mod.rs]: After discussion with @aleksuss we decided that `consensus key` is clearer, even it seems a little bit redundant.

File: cli/src/command/finalize.rs [hunk 6]
```diff
@@ -137,16 +137,16 @@ impl ExonumCommand for Finalize {
...
 
         if validators_count != public_configs.len() {
             bail!("The number of validators does not match the number of validators keys.");
         }
 
-        let genesis = GenesisConfig::new_with_consensus(
-            common.clone().consensus_config,
-            public_configs.iter().map(|c| c.validator_keys),
-        );
+        let genesis = ConsensusConfig {
```
Comment (vitvakatu) [inline cli/src/command/finalize.rs]: Why it is called genesis if it's ConsensusConfig?

File: examples/cryptocurrency/examples/demo.rs [hunk 7]
```diff
@@ -23,11 +23,13 @@ fn node_config() -> NodeConfig {
     let (consensus_public_key, consensus_secret_key) = exonum::crypto::gen_keypair();
     let (service_public_key, service_secret_key) = exonum::crypto::gen_keypair();
 
-    let validator_keys = ValidatorKeys {
-        consensus_key: consensus_public_key,
-        service_key: service_public_key,
+    let genesis = ConsensusConfig {
```
Comment (vitvakatu) [inline examples/cryptocurrency/examples/demo.rs]: Same question here

**Comment by popzxc** [comment]
Please add a changelog entry.

File: exonum/src/blockchain/config.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (alekseysidorov) [reply exonum/src/blockchain/config.rs]: It rather an interface detail.

**Pull request merged**